src/acrcssc/azext_acrcssc/_validators.py

```diff
@@ -0,0 +1,183 @@
+# pylint: disable=line-too-long
+# pylint: disable=broad-exception-caught
+# pylint: disable=logging-fstring-interpolation
+import json
+import os
+import re
+from knack.log import get_logger
+from azure.cli.command_modules.acr.repository import acr_repository_show
+from .helper._constants import (
+    BEARER_TOKEN_USERNAME,
+    CSSC_WORKFLOW_POLICY_REPOSITORY,
+    CONTINUOUSPATCH_IMAGE_LIMIT,
+    CONTINUOUSPATCH_OCI_ARTIFACT_CONFIG,
+    CONTINUOUSPATCH_CONFIG_SCHEMA_V1,
+    CONTINUOUSPATCH_CONFIG_SCHEMA_SIZE_LIMIT,
+    CONTINUOUSPATCH_ALL_TASK_NAMES,
+    ERROR_MESSAGE_INVALID_TIMESPAN_FORMAT,
+    ERROR_MESSAGE_INVALID_TIMESPAN_VALUE,
+    SUBSCRIPTION)
+from .helper._constants import CSSCTaskTypes, ERROR_MESSAGE_INVALID_TASK, RECOMMENDATION_SCHEDULE
+from .helper._ociartifactoperations import _get_acr_token
+from azure.mgmt.core.tools import (parse_resource_id)
+from azure.cli.core.azclierror import InvalidArgumentValueError
+from ._client_factory import cf_acr_tasks
+from .helper._utility import get_task
+
+logger = get_logger(__name__)
+
+
+def validate_continuouspatch_config_v1(config_path):
+    _validate_continuouspatch_file(config_path)
+    config = _validate_continuouspatch_json(config_path)
+    _validate_continuouspatch_config(config)
+
+
+def _validate_continuouspatch_file(config_path):
+    if not os.path.exists(config_path):
+        raise InvalidArgumentValueError(f"Config path file: {config_path} does not exist in the path specified")
+    if not os.path.isfile(config_path):
+        raise InvalidArgumentValueError(f"Config path file: {config_path} is not a valid file")
+    if os.path.getsize(config_path) > CONTINUOUSPATCH_CONFIG_SCHEMA_SIZE_LIMIT:
+        raise InvalidArgumentValueError(f"Config path file: {config_path} is too large. Max size limit is 10 MB")
+    if os.path.getsize(config_path) == 0:
+        raise InvalidArgumentValueError(f"Config path file: {config_path} is empty")
+    if not os.access(config_path, os.R_OK):
+        raise InvalidArgumentValueError(f"Config path file: '{config_path}' is not readable")
+
+
+def _validate_continuouspatch_json(config_path):
+    from jsonschema import validate
+    try:
+        with open(config_path, 'r') as f:
+            config = json.load(f)
+            validate(config, CONTINUOUSPATCH_CONFIG_SCHEMA_V1)
+            return config
+    except Exception as e:
+        logger.error(f"Error validating the continuous patch config file: {e}")
+        raise InvalidArgumentValueError("File used for --config is not a valid config JSON file. Use --help to see the schema of the config file.")
+    finally:
+        f.close()
+
+
+def _validate_continuouspatch_config(config):
+    if not isinstance(config, dict):
+        raise InvalidArgumentValueError("Config file is not a valid JSON file. Use --help to see the schema of the config file.")
+    for repository in config.get("repositories", []):
```

Comment:
Question, shall we throw the warning if 'repositories' not exist in the config or repositories list is empty?

Reply:
that is handled on the schema validation, it allows us to check that the arrays contain at least one element
https://github.com/AzureCR/azure-cli-extensions/blob/fc0beb69cbe54574219e1ac38f494257d0c04f21/src/acrcssc/azext_acrcssc/helper/_constants.py#L133